app/assets/javascripts/discourse/app/lib/text.js

```diff
@@ -23,6 +23,7 @@ function getOpts(opts) {
       formatUsername,
       watchedWordsReplace: context.site.watched_words_replace,
       watchedWordsLink: context.site.watched_words_link,
+      additionalOptions: context.site.markdown_additional_options,
```

Comment:
Are we able to just collapse `context.site.censored_regexp`, `context.site.custom_emoji_translation` etc into a single `context.site.markdown_options` object? It'll provide a cleaner interface for all the engines to consume additional options that they need from the site's context.

Reply:
@tgxworld I'd rather not move stuff around in the same PR, I just need to be unblocked on this chat stuff. I will do a follow up PR to move things around.